### PR TITLE
docs(authn/x509): Detail clientAuth config for SSL.

### PR DESCRIPTION
**Pull request by stewchen**

### CODE REVIEW

File: setup/security/authentication/x509/index.md [hunk 1]
```diff
@@ -101,13 +102,34 @@ Encoding with any another OID can be done by editing the openssl.conf.
...
+chain, as required by the Spring Security integration. 
+
+```
+hal config security api ssl edit --client-auth # Set to WANT or NEED
+```
+
+Set `client-auth` to `WANT` if you still want SSL connections to succeed even if the client doesn’t 
+provide a certificate. This is useful if you enable x509 with another authentication method like
+OAuth, LDAP, SAML - when a certificate is not provided, users can still authenticate with one of
+these methods.
```
Comment (dorbin) [inline setup/security/authentication/x509/index.md]: @stewchen , what does `WANT` add over not invoking this command at all?

File: setup/security/authentication/x509/index.md [hunk 2]
```diff
@@ -101,13 +102,34 @@ Encoding with any another OID can be done by editing the openssl.conf.
 
 ![Example x509 certificate generated](two_roles_x509.png)
 
-## Setting roleOid via halyard
+## Setting roleOid
```
Comment (dorbin) [inline setup/security/authentication/x509/index.md]: s/Setting/Set

File: setup/security/authentication/x509/index.md [hunk 3]
```diff
@@ -101,13 +102,34 @@ Encoding with any another OID can be done by editing the openssl.conf.
...
 
 ```
 hal config security authn x509 edit --role-oid 1.2.840.10070.8.1
 ```
 
-## Enabling x509 via halyard
+### Configuring SSL to require certs
```
Comment (dorbin) [inline setup/security/authentication/x509/index.md]: s/Configuring/Configure

File: setup/security/authentication/x509/index.md [hunk 4]
```diff
@@ -101,13 +102,34 @@ Encoding with any another OID can be done by editing the openssl.conf.
...
+
+Set `client-auth` to `NEED` if x509 is the sole authentication method, or if you want to ensure the
+certificate is provided AND another authentication mechanism is used.
+
+> TODO: add a hal command to support clearing --client-auth
+
+To revert back to not requiring a certificate after disabling x509, find and delete the `client-auth`
+field set in `~/.hal/config`
+
+## Enabling x509
```
Comment (dorbin) [inline setup/security/authentication/x509/index.md]: s/Enabling/Enable

**Review by dorbin — COMMENTED** [review]
LGTM, with those changes, and pending review by ttomsu

File: setup/security/authentication/x509/index.md (hunk 1 above)
Comment (stewchen) [reply setup/security/authentication/x509/index.md]: `WANT` will allow the server to use a certificate if available, whereas when this is unset, the server will not support certificates at all.

**Comment by ttomsu** [comment]
LGTM

**Review by ttomsu — APPROVED** [review]

**Pull request merged**